server/middlewares/authenticate.js

```diff
@@ -0,0 +1,37 @@
+const router = require('express').Router();
+const jwt = require('jsonwebtoken');
+const config = require('../config.js'); //holds jwtSecret
+
+const authenticate = (req, res, next) => {
+  const authorizationHeader = req.headers['authorization'];
+  console.log('this is request header: ', req.headers);
+  let token;
+  if (authorizationHeader) {
+    token = authorizationHeader.split(' ')[1];
+    console.log('has token');
+  }
+  console.log('herhehrehehreehr');
+
+  if(token) {
+    //validate
+    console.log('have token');
+    jwt.verify(token, config.jwtSecret, (err, decoded) => {
+      if (err) {
+        res.status(401).json({
+          error: 'Failed to authenticate'
+        });
+      } else {
```

Comment:
I think this is the point at where you could take the decoded credentials and add them to the request object so the 'next' handler can have access to them.